### PR TITLE
ENH: distutils: parallelize builds by default

### PR DESCRIPTION
**Pull request by ghost**
This may have some backcompat implications, but should be mostly positive.

### CODE REVIEW

File: numpy/distutils/misc_util.py [hunk 1]
```diff
@@ -92,7 +93,8 @@ def get_num_build_jobs():
 
     """
     from numpy.distutils.core import get_distribution
-    envjobs = int(os.environ.get("NPY_NUM_BUILD_JOBS", 1))
+    cpu_count = multiprocessing.cpu_count()
```
Comment (juliantaylor) [inline numpy/distutils/misc_util.py]: don't use cpu_count it doesn't respect affinity, use `len(os.sched_getaffinity(0))` when available.

Comment (xoviat) [reply numpy/distutils/misc_util.py]: Will do, but this should really be implemented in Python.

Comment (juliantaylor) [reply numpy/distutils/misc_util.py]: that is my opinion too, but the best I could get from them is a docstring update ...

File: numpy/distutils/misc_util.py [hunk 2]
```diff
@@ -92,7 +93,11 @@ def get_num_build_jobs():
 
     """
     from numpy.distutils.core import get_distribution
-    envjobs = int(os.environ.get("NPY_NUM_BUILD_JOBS", 1))
+    try:
+        cpu_count = len(os.sched_getaffinity(0))
+    except AttributeError:
+        cpu_count = multiprocessing.cpu_count()
```
Comment (charris) [inline numpy/distutils/misc_util.py]: Is this safe? Is it always faster?

Comment (xoviat) [reply numpy/distutils/misc_util.py]: Yes, it's safe and it's always faster (unless you're compiling with high CPU load and you have an OS scheduler that sucks, which is almost never the case nowadays).

Comment (xoviat) [reply numpy/distutils/misc_util.py]: high CPU load = gaming AND compiling numpy at the same time (I don't play games).

**Comment by charris** [comment]
Thanks @xoviat .

**Pull request merged**